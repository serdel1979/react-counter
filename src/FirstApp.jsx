import PropTypes from 'prop-types';

const cuadrado = (n)=>{
    return n*n;
}

export const FirstApp = ({title, subTitle, name})=>{ 

    if (!title){
        throw new Error('Debe enviar el t√≠tulo.');
    }

    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p> { name }</p>
        </>
    );
}

FirstApp.propTypes={
   title: PropTypes.string.isRequired,
   subtitle: PropTypes.string
}