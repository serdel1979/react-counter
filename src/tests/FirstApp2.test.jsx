import { FirstApp } from '../FirstApp'
const { render, screen } = require("@testing-library/react")


describe('Test en FirstApp', () => {

  const title = 'Hola, soy el título';
  const subTitle ='Soy el subtitulo';

  test('Debe hacer match con el snapshot', () => {
    
    const { container } = render(<FirstApp title={title}/>)
    expect( container ).toMatchSnapshot();
  })
  
  test('Debe mostrar el mensaje "Hola, soy el título" ', () => {
    render(<FirstApp title={title}/>);
    expect( screen.getByText(title)).toBeTruthy();

  })
  
  test('Debe mostrar el titulo en un h1', () => {
    render(<FirstApp title={title}/>);
    expect( screen.getByRole('heading',{level: 1}).innerHTML).toContain(title);
  })

  test('Debe mostrar el subtitulo', () => {
    render(<FirstApp title={title} subTitle = {subTitle}/>);
    expect( screen.getAllByText(subTitle).length).toBe(2);
})
  

})