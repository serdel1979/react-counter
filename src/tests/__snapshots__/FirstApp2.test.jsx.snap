// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy el t√≠tulo
  </h1>
  <p />
  <p />
  <p>
     
  </p>
</div>
`;
